CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
OK, so I've been trying to figure out why I hesitate to merge... and basically... I don't get it.

Specifically:
 * The PR and JIRA titles imply there are PHP 7.1 errors. What are the errors? How do you trigger them? (Maybe it seemed obvious to the you guys because they happen on all reports with all configurations in PHP 7.1? Or maybe they only happen sometimes? As a reader coming from first principles, I have no idea.)
 * It seems the final outcome of the `CRM_Report_Form::groupBy()`  is to fill the property `$this->_groupBy` by calling `getGroupByFromSelectColumns()` with our array. But `$groupBy` and `$this->_groupByArray` are initialized and built the same way. The difference is the side-effect of storing `_groupByArray`. But I can't figure anyway that this side-effect makes a difference: nobody else reads `$this->_groupByArray`; you suggest that most reports would only execute it once; the child class mentioned here doesn't use (`CRM_Report_From::groupBy()`). 
    * Maybe the change in `CRM_Report_Form::groupBy()` is just a style-change to set a tone? Maybe the other change (`CRM_Report_Form_Member_Summary::groupBy()`) is the meaningful one?